frontend/src/v5/ui/routes/dashboard/projects/federations/federationSettingsForm/federationSettingsForm.component.tsx

```diff
@@ -0,0 +1,272 @@
+import { FormTextField } from '@controls/formTextField/formTextField.component';
+import { FormSelectView } from '@controls/formSelectView/formSelectView.component';
+import { FormSelect } from '@controls/formSelect/formSelect.component';
+import { FederationSettingsSchema } from '@/v5/validation/schemes';
+import { FormTextFieldUnit } from '@controls/formTextFieldUnit/formTextFieldUnit.component';
+import { FlexContainer, SectionTitle, ShareTextField } from './federationSettingsForm.styles';
+
+const UNITS = [
+	{
+		name: formatMessage({ id: 'units.mm.name', defaultMessage: 'Millimetres' }),
+		abbreviation: formatMessage({ id: 'units.mm.abbreviation', defaultMessage: 'mm' }),
+	},
+	{
+		name: formatMessage({ id: 'units.cm.name', defaultMessage: 'Centimetres' }),
+		abbreviation: formatMessage({ id: 'units.cm.abbreviation', defaultMessage: 'cm' }),
+	},
+	{
+		name: formatMessage({ id: 'units.dm.name', defaultMessage: 'Decimetres' }),
+		abbreviation: formatMessage({ id: 'units.dm.abbreviation', defaultMessage: 'dm' }),
+	},
+	{
+		name: formatMessage({ id: 'units.m.name', defaultMessage: 'Metres' }),
+		abbreviation: formatMessage({ id: 'units.m.abbreviation', defaultMessage: 'm' }),
+	},
+	{
+		name: formatMessage({ id: 'units.ft.name', defaultMessage: 'Feet and inches' }),
+		abbreviation: formatMessage({ id: 'units.ft.abbreviation', defaultMessage: 'ft' }),
+	},
+];
+
+const DECIMAL_UNIT = UNITS[2];
+
+interface IFormInput {
+	name: string;
+	unit: string;
+	desc: string;
+	code: string;
+	defaultView: string;
+	latitude: number;
+	longitude: number;
+	angleFromNorth: number;
+	x: number;
+	y: number;
+	z: number;
+}
+
+const getDefaultValues = (federation: IFederation) => {
+	const DEFAULT_UNIT = UNITS[0];
+	const {
+		unit = DEFAULT_UNIT.abbreviation,
+		angleFromNorth = 0,
+		code,
+		name,
+		desc = '',
+	} = federation;
+	const defaultView = federation.defaultView || EMPTY_VIEW._id;
+	const { latLong, position } = federation.surveyPoint || {};
+	const [x, y, z] = position || [0, 0, 0];
+	const [latitude, longitude] = latLong || [0, 0];
+	return {
+		name,
+		desc,
+		code,
+		unit,
+		defaultView,
+		latitude,
+		longitude,
+		angleFromNorth,
+		x,
+		y,
+		z,
+	};
+};
+
+type IFederationSettingsForm = {
+	open: boolean;
+	federation: IFederation;
+	onClose: () => void;
+};
+
+export const FederationSettingsForm = ({ open, federation, onClose }: IFederationSettingsForm) => {
+	let defaultValues = getDefaultValues(federation) as any;
+	const {
+		handleSubmit,
+		reset,
+		watch,
+		control,
+		formState,
+		formState: { errors },
+	} = useForm<IFormInput>({
+		mode: 'onChange',
+		resolver: yupResolver(FederationSettingsSchema),
+		defaultValues,
+	});
+
+	const currentUnit = watch('unit');
+
+	const { teamspace, project } = useParams() as { teamspace: string, project: string };
+
+	useEffect(() => {
+		defaultValues = getDefaultValues(federation) as any;
```

Comment:
Can we avoid fetching the federation settings and views if open === false?
```suggestion
		if (!open) {
				return;
		}
		defaultValues = getDefaultValues(federation) as any;
```